perfkitbenchmarker/scripts/step_load_driver/launch_mysql_service.py

```diff
@@ -0,0 +1,258 @@
+                      'collected, in seconds.')
+gflags.DEFINE_list(THREAD_COUNT_LIST, [1, 2, 5, 8],
+                   'The number of test threads on the client side.')
+gflags.DEFINE_integer(SYSBENCH_REPORT_INTERVAL, 2,
+                      'The interval, in seconds, we ask sysbench to report '
+                      'results.')
+gflags.DEFINE_string(RUN_URI, None,
+                     'Run identifier, if provided, only run phase '
+                     'will be completed.')
+gflags.DEFINE_string(STORAGE_BUCKET, None,
+                     'GCS bucket to upload records to. Bucket must exist.')
+gflags.DEFINE_list(ADDITIONAL_FLAGS, None,
+                   'List of additional PKB mysql_service valid flags (strings).'
+                   'For example: ["--storage_size=100"].')
+
+
+class TimeOutError(Exception):
+  pass
+
+
+def driver(argv):
+  """Driver initiates sysbench run with different thread counts.
+
+  Args:
+    argv: system arguments (command line flags).
+  """
+  try:  # Parse command line flags
+    argv = FLAGS(argv)
+  except gflags.FlagsError as e:
+    logging.error('%s\nUsage: %s ARGS\n%s', e, sys.argv[0], FLAGS)
+    sys.exit(1)
+  run_uri = FLAGS.run_uri
+  if not run_uri:
+    logging.info('No run_uri given. Will run full sysbench test.')
+    run_uri = _provision_prepare_pkb()
+    logging.info('Provision and prepare completed. Run uri assigned: %s',
+                 run_uri)
+  _run(run_uri)
+  if not FLAGS.run_uri:
+    _cleanup_teardown_pkb(run_uri)
+
+
+def _provision_prepare_pkb():
+  """Run cleanup stage of PKB benchmark.
+
+  Returns:
+    run_uri: (string)
+  """
+  pkb_cmd = PKB + STAGE_FLAG + PROVISION + ',' + PREPARE
+  if FLAGS.additional_flags:
+    pkb_cmd = _append_additional_flags(pkb_cmd)
+  # PKB run with prepare,provision, wait
+  logging.info('Provision and prepare sysbench with the following command:\n%s',
+               pkb_cmd)
+  [stdout_filename, stderr_filename] = _generate_filenames(PROVISION, None)
+  _execute_pkb_cmd(pkb_cmd, stdout_filename, stderr_filename)
+  return _wait_for_run(stderr_filename)
+
+
+def _run(run_uri):
+  """Run stage of PKB benchmark.
+
+  Args:
+    run_uri: (string).
+  """
+
+  run_iterations = len(FLAGS.thread_count_list)
+  logging.info(
+      'Beginning run phase. Will execute runs with %d different thread counts.',
+      run_iterations)
+  for t in FLAGS.thread_count_list:
+    pkb_cmd = (PKB + STAGE_FLAG + RUN + URI_FLAG + run_uri + THREAD_FLAG +
+               str(t) + RUN_TIME + str(FLAGS.sysbench_run_seconds) + ' &')
+    if FLAGS.additional_flags:
+      pkb_cmd = _append_additional_flags(pkb_cmd)
+    stdout_filename, stderr_filename = _generate_filenames(RUN, t)
+    logging.info('Executing PKB run with thread count: ' + str(t))
+    _execute_pkb_cmd(pkb_cmd, stdout_filename, stderr_filename)
+    _wait_for_run(stderr_filename)
+    logging.info('Finished executing PKB with thread count: ' + str(t))
```

Comment:
str(t): here, below and above - is it really necessary to do str(integer) transition?

Typical convention as seen in the rest of PKB code base for string formatting is following:

string_result = "template string: %d and %s" % (123, "xyz")

this will put 123 in place of %d and "xyz" in place of %s

So for this line, it'd look like "finished executing... %d" % t

But meanwhile, I can see that str(t) fits naturally when you need to concatenate multiple strings together using + sign, it is also quite readable.

So your call here: whether you want to convert some of the str(t) here to fit more consistently with the rest of the PKB code base.